pilot/pkg/features/pilot.go

```diff
@@ -135,6 +135,12 @@ var (
 		false,
 		"If enabled, controller that untaints nodes with cni pods ready will run. This should be enabled if you disabled ambient init containers.").Get()
 
+	EnableV2IPAutoallocate = env.Register(
```

Comment:
Since this is a 'v2' API - can we add a more granular option to enable this ? 

For example a label on the ServiceEntry - having a global 'all your ServiceEntries get IPs no matter what' will break or change in subtle ways other features where we handle SE that don't have an IP. 

For example if a user is mostly using SE for clusters (route destination) - it would create a large number of listeners and change HTTP and other behaviors.

Instead: add AutoallocateLabel name, and if it is set any SE with that label will get allocated addresses.

Reply:
In context of the other comments - I would just remove this option, instead add the 2 CIDR range options which are equivalent indication the user wants to allocate, plus the label to opt in specific ServiceEntries.

I don't want to get into a debate about 'all users will want this feature and associated DNS interception, so it must be on by default with out-out' - if we do find out that a majority of users enable this we can change it, as of now we have no evidence one way or another and the safest option is to start with opt-in. Much easier to make something that is 'opt-in' default with an opt-out option than to make something that is 'on by default with opt-out' an opt-in.